app/models/application_form.rb

```diff
@@ -411,6 +412,16 @@ def reviewable?(section)
     apply_2? && previous_application_rejection_reason(section).present?
   end
 
+  def self.with_unsafe_application_choice_touches
+    prior_state = RequestStore.store[:allow_unsafe_application_choice_touches].presence || false
+
+    RequestStore.store[:allow_unsafe_application_choice_touches] = true
+    return_value = yield
```

Comment:
Not entirely sure what this 'yeild' does

Reply:
yeah — it's obscure! `yield` runs the block that was passed to the method and returns what the block returns. so if we call 

```ruby
ApplicationForm. with_unsafe_application_choice_touches do
  create(:application_choice)
end
```

`yield` will run the block and return the created application choice

Reply:
That makes a lot of sense, in which case I'm happy to approve =)